docs/source/development.md

```diff
@@ -0,0 +1,93 @@
+# Overview
+
+Information for contributors
+
+## Quick start
+* Change current directory to the top level of your local Forest repository
+* Install Forest in editable mode: `pip install -e .`
+* Install development tools: `pip install -r requirements.txt`
+* Run code style checks: `flake8`
+* Run type hint checks: `mypy -p forest`
+* Run test suite: `pytest`
+
+## General information
+* [Forest Naming Conventions](naming.md)
+* [Exception handling](exceptions.md)
+* [Logging](logging.md)
+* Delete unnecessary code (don't just comment it out)
+* [Write short functions that do one thing only](https://github.com/amontalenti/elements-of-python-style#if-the-implementation-is-hard-to-explain-its-a-bad-idea)
+* Specify exact version numbers for dependencies (pin your requirements)
+* Refactor the code and avoid copy-paste programming ([Don't Repeat Yourself](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself))
+* Write tests for all new code
+
+## Python
+* Follow [PEP 8](https://www.python.org/dev/peps/pep-0008/), [The Zen of Python](https://www.python.org/dev/peps/pep-0020/), [Google style guide for docstrings](https://google.github.io/styleguide/pyguide.html#s3.8.1-comments-in-doc-strings), and [The Elements of Python Style](https://github.com/amontalenti/elements-of-python-style) guidelines
+* Avoid one character names (except with `lambda`, generator expressions, and with unpacking when using `_` as a throwaway character)
+* Use double quotes for strings and use single quotes only to avoid backslashes in the string
+* Alphabetize imports within blocks
+* Use parentheses for [continuations](https://github.com/amontalenti/elements-of-python-style#use-parens--for-continuations) and [method chaining](https://github.com/amontalenti/elements-of-python-style#use-parens--for-fluent-apis)
+* Put a line break before a binary operator
+* Use the [logging module](https://docs.python.org/3/library/logging.html) instead of `print()`
+* [Formatting log messages](http://reinout.vanrees.org/weblog/2015/06/05/logging-formatting.html): `logger.info("%s went %s wrong", 42, 'very')`
+* Continue writing code and comments all the way until the end of the line then indent appropriately
+
+## GitHub
+
+Development workflow:
+1. open an issue
+1. create a pull request
+1. merge pull request
+1. close issue
+
+### Branches
+
+* Follow the [Gitflow Workflow](https://www.atlassian.com/git/tutorials/comparing-workflows/gitflow-workflow) for managing branches
+* Use descriptive names for branches
+* Keep branches short-lived to avoid potential merge conflicts
+
+### Pull requests
+* 
+* PRs:
+  * are not just peer review but a way to communicate and learn about the code base
+  * help build and pass on institutional knowledge, and keep default branch commit history clean
+* Keep PRs small (ideally, under 100 lines of code) to help makes reviews easier and faster
+* Don't open a pull request for a pull request
```

Comment:
It means don't create a feature branch from another feature branch to open a PR on top of an existing PR. Simply push updates to the existing feature branch and re-request the review of the existing PR. What would be a succinct way to express this?

Reply:
Hmm. Is this clear? 

"If you are adding changes to a branch which has an open pull request, add commits to the existing pull request rather than creating a new pull request."